core/src/avm1/script_object.rs

```diff
@@ -0,0 +1,702 @@
+use crate::avm1::function::{Executable, NativeFunction};
+use crate::avm1::property::{Attribute, Property};
+use crate::avm1::return_value::ReturnValue;
+use crate::avm1::{Avm1, Error, Object, ObjectCell, UpdateContext, Value};
+use crate::display_object::DisplayNode;
+use core::fmt;
+use enumset::EnumSet;
+use gc_arena::{GcCell, MutationContext};
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+
+pub const TYPE_OF_OBJECT: &str = "object";
+pub const TYPE_OF_FUNCTION: &str = "function";
+pub const TYPE_OF_MOVIE_CLIP: &str = "movieclip";
+
+#[derive(Clone)]
+pub struct ScriptObject<'gc> {
+    prototype: Option<ObjectCell<'gc>>,
+    display_node: Option<DisplayNode<'gc>>,
+    values: HashMap<String, Property<'gc>>,
+    function: Option<Executable<'gc>>,
+    type_of: &'static str,
+}
+
+unsafe impl<'gc> gc_arena::Collect for ScriptObject<'gc> {
+    fn trace(&self, cc: gc_arena::CollectionContext) {
+        self.prototype.trace(cc);
+        self.display_node.trace(cc);
+        self.values.trace(cc);
+    }
```

Comment:
Missing trace on `self.function`?

Reply:
Yes. Interestingly enough I caught this in a child branch that I haven't even PRd yet. Fix upcoming in the next rebase